ui/src/components/ui/device-control-panel/tabs/dashboard-tab.tsx

```diff
@@ -58,13 +69,24 @@ export const DashboardTab = () => {
         Stub
       </DeviceControlCard>
       <DeviceControlCard
-        after={<Button appearance='neutral' before={<Icon20AddSquareOutline />} mode='tertiary' />}
         afterTooltipText={t('Extend booking')}
         before={<Icon28StopwatchOutline height={20} width={20} />}
         title={t('Device booking')}
+        after={
+          <Button
+            appearance='neutral'
+            before={<Icon20AddSquareOutline />}
+            mode='tertiary'
+            onClick={() => {
+              if (!serial) return
```

Comment:
I see this line more often that I think I would like.
Could we either make sure that at this point in the render tree we definitely have a serial number and make it not null, or make a context for when we are working with a device?

These components don't make sense without a specified device, so serial should be not null. I'm not sure what's the best way to implement it though

Reply:
Maybe instead of reading serial from a url, we supply it using a context that reads the params and redirects to the device list page if the device is not found?